base/linalg/exceptions.jl

```diff
@@ -29,10 +29,30 @@ function Base.showerror(io::IO, ex::ARPACKException)
     end
 end
 
+"""
+   SingularException(info)
+
+The matrix passed is a singular matrix. Optional argument INFO is an INTEGER > 0  
+      if INFO = i, The matrix is exactly singular, and division by zero 
+      will occur if it is used to solve a system of equations.
```

Comment:
We'll need to add here that when pivoting is disabled for a factorization which exception might not indicate that matrix is singular. E.g. the matrix `[0 1l 1 1]` will raise this exception for LU without pivoting. Also, please avoid copying the Fortran variable and type names from the LAPACK documentation. `INTEGER` is not really a thing in Julia and the variable in Julia is `info` not `INFO`.

Reply:
I want to cross check if I understood it correctly, "if pivoting is disabled the value of `info` is irrelevant to the singularity of the matrix" is this what you meant? Sorry for the not paying attention to `INTEGER` !